src/command/backup/backup.c

```diff
@@ -0,0 +1,1906 @@
+                                manifestData(manifest)->backupLabelPrior ?
+                                    strPtr(manifestData(manifest)->backupLabelPrior) : "<undef>");
+                        }
+                        // Check compression. Compression can't be changed between backups so resume won't work either.
+                        else if (manifestResumeData->backupOptionCompress != cfgOptionBool(cfgOptCompress))
+                        {
+                            reason = strNewFmt(
+                                "new compression '%s' does not match resumable compression '%s'",
+                                cvtBoolToConstZ(cfgOptionBool(cfgOptCompress)),
+                                cvtBoolToConstZ(manifestResumeData->backupOptionCompress));
+                        }
+                        else
+                            usable = true;
+                    }
+                    CATCH_ANY()
+                    {
+                    }
+                    TRY_END();
+                }
+
+                // If the backup is usable then return the manifest
+                if (usable)
+                {
+                    result = manifestMove(manifestResume, MEM_CONTEXT_OLD());
+                }
+                // Else warn and remove the unusable backup
+                else
+                {
+                    LOG_WARN_FMT("backup '%s' cannot be resumed: %s", strPtr(backupLabel), strPtr(reason));
+
+                    storagePathRemoveP(
+                        storageRepoWrite(), strNewFmt(STORAGE_REPO_BACKUP "/%s", strPtr(backupLabel)), .recurse = true);
+                }
+            }
+            }
+        }
+    MEM_CONTEXT_TEMP_END();
+
+    FUNCTION_LOG_RETURN(MANIFEST, result);
+}
+
+static bool
+backupResume(Manifest *manifest, const String *cipherPassBackup)
+{
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+        FUNCTION_LOG_PARAM(MANIFEST, manifest);
+        FUNCTION_TEST_PARAM(STRING, cipherPassBackup);
+    FUNCTION_LOG_END();
+
+    ASSERT(manifest != NULL);
+
+    bool result = false;
+
+    MEM_CONTEXT_TEMP_BEGIN()
+    {
+        const Manifest *manifestResume = backupResumeFind(manifest, cipherPassBackup);
+
+        // If a resumable backup was found set the label and cipher subpass
+        if (manifestResume)
+        {
+            // Resuming
+            result = true;
+
+            // Set the backup label to the resumed backup
+            manifestBackupLabelSet(manifest, manifestData(manifestResume)->backupLabel);
+
+            LOG_WARN_FMT(
+                "resumable backup %s of same type exists -- remove invalid files and resume",
+                strPtr(manifestData(manifest)->backupLabel));
+
+            // If resuming a full backup then copy cipher subpass since it was used to encrypt the resumable files
+            if (manifestData(manifest)->backupType == backupTypeFull)
```

Comment:
If incr/diff we need to get cipher pass from the prior backup (which has already been done).  If full, we need it from the resumed manifest.  If we exclude this check it would still work, but we'd set the cipher pass twice for diff/incr backups which seems weird.